tools/fasta_compute_length/utils/fasta_to_len.py

```diff
@@ -25,28 +22,28 @@ def compute_fasta_length( fasta_file, out_file, keep_first_char, keep_first_word
         keep_first_char += 1
 
     first_entry = True
-
-    for line in open( infile ):
-        line = line.strip()
-        if not line or line.startswith( '#' ):
-            continue
-        if line[0] == '>':
-            if first_entry == False:
-                if keep_first_word:
-                    fasta_title = fasta_title.split()[0]
-                out.write( "%s\t%d\n" % ( fasta_title[ 1:keep_first_char ], seq_len ) )
+    with open(fasta_file) as in_fh, open(out_file, 'w') as out_fh:
+        for line in in_fh:
+            line = line.strip()
+            if not line or line.startswith('#'):
+                continue
+            if line[0] == '>':
+                if first_entry is False:
+                    if keep_first_word:
+                        fasta_title = fasta_title.split()[0]
+                    out_fh.write("%s\t%d\n" % (fasta_title[1:keep_first_char], seq_len))
+                else:
+                    first_entry = False
+                fasta_title = line
```

Comment:
Why do you think the tool doesn't work if there is no space after `>` ?

Reply:
I have to correct my self: I think it "fails" if there is a space after `>` because 

https://github.com/mvdbeek/tools-devteam/blob/ab3b801033ce62a0084f1d91a56a87e55c262fa9/tools/fasta_compute_length/utils/fasta_to_len.py#L33

would take the `>` as `fasta_title`, or?

Reply:
it would, but at this point it's already too late and would only affect the last fasta entry. We could check if `fasta_title == >` and then adjust accordingly, but for my taste that's a bit too much magic. We could error out with an `assert fasta_title != '>'`